ansible/roles/infra-ec2-template-generate/tasks/main.yml

```diff
@@ -60,44 +69,4 @@
         src: "{{ cloudformation_template }}"
         mode: put
 
-######################### Validate CF Template
-
-- name: validate cloudformation template (local)
-  environment:
-    AWS_ACCESS_KEY_ID: "{{aws_access_key_id}}"
-    AWS_SECRET_ACCESS_KEY: "{{aws_secret_access_key}}"
-    AWS_DEFAULT_REGION: "{{aws_region_final|d(aws_region)}}"
-  command: >-
-    aws cloudformation validate-template
-    --region {{ aws_region_final | d(aws_region) | default(region) | default('us-east-1')}}
-    --template-body file://{{ cloudformation_template }}
-  changed_when: false
-  register: cloudformation_validation
-  until: cloudformation_validation is succeeded
-  retries: "{{ cloudformation_retries }}"
-  delay: 20
-  tags:
-    - aws_infrastructure_deployment
-    - validate_cf_template
-  when: stat_template.stat.size <= 51200
-
-- name: validate cloudformation template (S3)
-  environment:
-    AWS_ACCESS_KEY_ID: "{{aws_access_key_id}}"
-    AWS_SECRET_ACCESS_KEY: "{{aws_secret_access_key}}"
-    AWS_DEFAULT_REGION: "{{aws_region_final|d(aws_region)}}"
-  command: >-
-    aws cloudformation validate-template
-    --region {{ aws_region_final | d(aws_region) | default(region) | default('us-east-1')}}
-    --template-url https://s3.amazonaws.com/{{bucket_templates}}/{{env_type}}.{{guid}}.{{cloud_provider}}_cloud_template
-  changed_when: false
-  register: cloudformation_validation
-  until: cloudformation_validation is succeeded
-  retries: "{{ cloudformation_retries }}"
-  delay: 20
-  tags:
-    - aws_infrastructure_deployment
-    - validate_cf_template
-  when: stat_template.stat.size > 51200
-
```

Comment:
Simple validation of yaml / json  is not a full replacement for the validate-template capability provided by the CLI.

You cannot just delete this.

What about having a boolean, `cloudformation_validation` (or pick an appropriate name) that you could enable / disable for you CI ?

Reply:
Sure, you are right, I'll update it. I was just looking at the docs to [validate-template](https://docs.aws.amazon.com/cli/latest/reference/cloudformation/validate-template.html), and this change came in mind. They say it is just a JSON/YAML validation.

Reply:
I think `validate-template` catches other things like:

> A client error (ValidationError) occurred when calling the ValidateTemplate operation: Template format error: Every Outputs member must contain a Value object

Reply:
Ok, than with the last change the functionality is preserved.

Reply:
What do you think about the change? Is it ok now? CI now has some kind of checks, and in order to skip the failing part, tags are used. @fridim